README.md

```diff
@@ -11,49 +11,171 @@
+#### Parameters
+#### `octokit`
+*octokit* <br>
+**Required.** An authenticated [octokit](https://github.com/octokit/rest.js/) instance
 
-**Table of contents:**
+#### `changes`
+*Map<string, FileData>* <br>
+**Required.** A set of files with their respective file contents conforming where the key is the file path, and the value is the a FileData object.
 
+**FileData Object**
+|  field 	|   type	|   description	|
+|---	|---	|---	|
+|   mode	|   `'100644' \| '100755' \| '040000' \| '160000' \| '120000'`	|  The file type as specified in the [GitHub API](https://developer.github.com/v3/git/trees/#tree-object). Default is `'100644'`. From the docs: "The file mode; one of 100644 for file (blob), 100755 for executable (blob), 040000 for subdirectory (tree), 160000 for submodule (commit), or 120000 for a blob that specifies the path of a symlink."|
+|   content	|  `string | null` 	|  **Required.** The entire file contents  	|
 
-* [Quickstart](#quickstart)
+#### `config`
+*Config Object* <br>
+**Required.** Descriptive values or enforced rules for pull requests, branching, and commits.
 
-  * [Installing the client library](#installing-the-client-library)
+**Config Object**
+|      field      |     type  	|   description	|
+|---------------	|-----------	|-------------	|
+|   upstreamRepo	|   `string`	| **Required.** The repository to suggest changes to.  |
+|   upstreamOwner	|   `string`	| **Required.** The owner of the upstream repository. |
+|   description	  |   `string`	| The GitHub Pull Request description. Default is `'code suggestions'`.  |
+|   title       	|   `string`	| The GitHub Pull Request title. Default is `'chore: code suggestions'`.      |
+|   branch	      |   `string`	| The branch containing the changes. Default is `'code-suggestions'`.   |
+|   primary	      |   `string`	| The primary upstream branch to open a PR against. Default is `'master'`.   |
+|   message     	|   `string`	| The commit message for the changes. Default is `'code suggestions'`. We recommend following [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/).|
+|   force	        |   `boolean`	| Whether or not to force push the reference even if the ancestor commits differs. Default is `false`. |
 
 
-* [Versioning](#versioning)
-* [Contributing](#contributing)
-* [License](#license)
 
-## Quickstart
+#### `logger`
+*[Logger](https://www.npmjs.com/package/@types/pino)* <br>
+The default logger is [Pino](https://github.com/pinojs/pino). You can plug in any logger that conforms to [Pino's interface](https://www.npmjs.com/package/@types/pino)
 
-### Installing the client library
+
+#### Exceptions
+The core-library will throw an exception if the [GitHub V3 API](https://developer.github.com/v3/) returns an error response, or if the response data format did not come back as expected. <br>
+
+## CLI
+
+### Installation
 
 ```bash
-npm install @google-cloud/code-suggester
+npm i code-suggester -g
```

Comment:
Generally `npm install -g` is considered poor form these days.  If The user is using this as a package that will be loaded at runtime in their application, `npm install code-suggester` will work.  If this is a tool they will use but not run at runtime, they would `npm install --save-dev code-suggester`, or `npx code-suggester` to do the initial work.